frontend/src/components/EmailSubscriptionContainer.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { subscribeToEmail } from '../services/reviewService';
+
+const EmailSubscriptionContainer: React.FC<{ email: string; setEmail: (email: string) => void; onSubmissionSuccess: () => void; }> = ({ email, setEmail, onSubmissionSuccess }) => {
```

Comment:
It's not the best thing to move the email input state outside of this component, since this component is specifically responsible of managing the email and signal the submission to the parent ThankYou component. I would keep email state inside of this component and change the prop handler from `onSubmissionSuccess` to `emailSubmissionHandler(email: string) => void`. In this way the component handles the input/validation logic and delegates the actual submission to the parent. Also, the component is not interested in the submission outcome.

Reply:
Fixed this